torch_geometric/nn/encoding.py

```diff
@@ -94,3 +96,182 @@ def forward(self, x: Tensor) -> Tensor:
+        h = self.channel_lin_2(h)
+        h = torch.nn.functional.dropout(h, p=0.5, training=self.training)
+        h_channel = h + h_token
+
+        # head
+        h_channel = self.head_layer_norm(h_channel)
+        t = torch.mean(h_channel, dim=1)
+        return self.head_lin(t)
+
+    def __repr__(self) -> str:
+        return (f"{self.__class__.__name__}("
+                f"num_tokens={self.num_tokens}, "
+                f"in_channels={self.in_channels}, "
+                f"out_channels={self.out_channels})")
+
+
+class LinkEncoding(torch.nn.Module):
+    r"""The link-encoding function from the `"Do We Really Need Complicated
+    Model Architectures for Temporal Networks?"
+    <https://openreview.net/forum?id=ayPPc0SyLv1>`_ paper.
+    :class:`LinkEncoding` is composed of two components. The first component is
+    :class:`TemporalEncoding` that maps each edge timestamp to a
+    ``time_channels`` dimensional vector.
+    The second component, 1-layer MLP-mixer, maps each encoded timestamp
+    feature concatenated with its corresponding link feature to a
+    ``out_channels`` dimensional vector.
+
+    Args:
+        K (int): The number of most recent teomporal links to use to construct
+            an intermediate feature representation for each node.
+        in_channels (int): The number of edge features used to construct
+            a linear layer encoding the output of :class:`TemporalEncoding`
+            concatenated with ``edge_attr``.
+        time_channels (int): dims to encode each timestamp into with
+            :class:`TemporalEncoding`.
+        out_channels (int): Size of each output sample.
+        is_sorted (bool, optional): If set to :obj:`True`, assumes that
+            :obj:`edge_index` is sorted by column. This avoids internal
+            re-sorting of the data and can improve runtime and memory
+            efficiency. (default: :obj:`False`)
+
+    Example:
+
+        >>> # GraphMixer paper uses the following args for GDELTLite dataset
+        >>> link_encoder = LinkEncoding(
+        ...     K=30,
+        ...     in_channels=186,
+        ...     hidden_channels=100,
+        ...     out_channels=100,
+        ...     time_channels=100,
+        ... )
+
+    """
+    def __init__(
+        self,
+        K: int,
+        in_channels: int,
+        hidden_channels: int,
+        out_channels: int,
+        time_channels: int,
+        is_sorted: bool = False,
+    ) -> None:
+        super().__init__()
+        self.K = K
+        self.in_channels = in_channels
+        self.hidden_channels = hidden_channels
+        self.out_channels = out_channels
+        self.time_channels = time_channels
+        self.is_sorted = is_sorted
+
+        # teomporal encoder
+        self.temporal_encoder = TemporalEncoding(time_channels)
+        self.temporal_encoder_head = torch.nn.Linear(
+            time_channels + in_channels,
+            hidden_channels,
+        )
```

Comment:
@weihua916 Just FYI, this linear layer wasn't mentioned in the paper AFAIK, but I added it to match this implementation with their official code: https://github.com/CongWeilin/GraphMixer/blob/bbdd9923706a02d0619dfd00ef7880911f003a65/model.py#LL170C25-L170C25

Reply:
Makes sense. Thanks!